pymysql/connections.py

```diff
@@ -1498,19 +1499,24 @@ def _get_descriptions(self):
                     # This behavior is different from TEXT / BLOB.
                     # We should decode result by connection encoding regardless charsetnr.
                     # See https://github.com/PyMySQL/PyMySQL/issues/488
-                    encoding = conn_encoding  # SELECT CAST(... AS JSON) 
-                elif field_type in TEXT_TYPES:
-                    if field.charsetnr == 63:  # binary
-                        # TEXTs with charset=binary means BINARY types.
-                        encoding = None
-                    else:
-                        encoding = conn_encoding
+                    encoding = conn_encoding  # SELECT CAST(... AS JSON)
+                elif field_type in TEXT_TYPES and field.charsetnr != 63:  # binary
+                    encoding = conn_encoding
                 else:
                     # Integers, Dates and Times, and other basic data is encoded in ascii
                     encoding = 'ascii'
             else:
                 encoding = None
-            converter = self.connection.decoders.get(field_type)
+
+            if field_type in TEXT_TYPES and field.charsetnr == 63:  # binary
+                # TEXTs with charset=binary means BINARY types.
+                encoding = None
+                isbinary = True
+
+            if not isbinary:
+                converter = self.connection.decoders.get(field_type)
+            else:
+                converter = None
```

Comment:
Why isbinary needed?  Why is this wrong?
Anyway, test for use_unicode=False is needed.

```py
            if field_type in TEXT_TYPES and field.charsetnr == 63:  # binary
                # TEXTs with charset=binary means BINARY types.
                encoding = None
                converter = None
            else:
                converter = self.connection.decoders.get(field_type)
```